service-base/src/main/java/org/eclipse/hono/service/deviceconnection/BaseDeviceConnectionService.java

```diff
@@ -18,167 +18,38 @@
 
 import org.eclipse.hono.client.ClientErrorException;
 import org.eclipse.hono.client.ServerErrorException;
-import org.eclipse.hono.service.EventBusService;
-import org.eclipse.hono.tracing.TracingHelper;
-import org.eclipse.hono.util.EventBusMessage;
-import org.eclipse.hono.util.MessageHelper;
-import org.eclipse.hono.util.DeviceConnectionConstants;
 import org.eclipse.hono.util.DeviceConnectionResult;
 
-import io.opentracing.References;
 import io.opentracing.Span;
-import io.opentracing.SpanContext;
-import io.opentracing.Tracer;
-import io.opentracing.tag.Tags;
 import io.vertx.core.AsyncResult;
 import io.vertx.core.Future;
 import io.vertx.core.Handler;
 
 /**
  * A base class for implementing {@link DeviceConnectionService}.
  * <p>
- * In particular, this base class provides support for receiving service invocation request messages
+ * This base class provides support for receiving service invocation request messages
  * via vert.x' event bus and routing them to specific methods accepting the
  * query parameters contained in the request message.
  *
- * @param <T> The type of configuration properties this service requires.
+ * @param <T> The type of configuration class this service supports.
+ * @deprecated - Use {@link DeviceConnectionService} instead.
  */
-public abstract class BaseDeviceConnectionService<T> extends EventBusService<T> implements DeviceConnectionService {
-
-    private static final String SPAN_NAME_GET_LAST_GATEWAY = "get last known gateway";
-    private static final String SPAN_NAME_SET_LAST_GATEWAY = "set last known gateway";
+@Deprecated
```

Comment:
I am not sure if we really need this class. It is deprecated, but we also didn't have this in the last release.

Reply:
The `BaseDeviceConnectionService` class is used in the `Application` class of the legacy device registry implementation.
I think it's more straightforward using the deprecated class there in the same fashion as the other deprecated service base classes.

Reply:
I was assuming to use the new pattern in the legacy device registry as well. Its just a bit weird that we add a new and immediately deprecated feature.

But yes, it might be good to simply repeat the pattern.